File: 프로그래머스/lv1/42748. K번째수/K번째수.py
def solution(array, command):
    result = []

    for i in range(len(command)):
        num = array[command[i][0] - 1:command[i][1]]
        num.sort()
        final = num[command[i][2] - 1]
        result.append(final)

    return result

File: 프로그래머스/lv1/42748. K번째수/K번째수.java


import java.util.*;
class Solution {
    public int[] solution(int[] array, int[][] commands)
    {
        int[] answer = new int[commands.length];
        // array배열을 자른 list
        ArrayList<Integer> list = new ArrayList<Integer>();

        for (int i = 0; i < commands.length; i++)
        {
            // array 숫자 배열 자르기
            for (int j = commands[i][0]; j <= commands[i][1]; j++)
            {
                list.add(array[j-1]);
            }
            // 오름차순 정렬
            Collections.sort(list);
            answer[i] = list.get(commands[i][2] - 1);
            list.clear();
        }
        return answer;
    }
}